VirtoCommerce.Storefront/Infrastructure/Autorest/BaseAuthHandler.cs

```diff
@@ -0,0 +1,53 @@
+using System.Net.Http;
+using Microsoft.AspNetCore.Http;
+using VirtoCommerce.Storefront.Model;
+
+namespace VirtoCommerce.Storefront.Infrastructure.Autorest
+{
+    public class BaseAuthHandler : DelegatingHandler
+    {
+        private readonly IWorkContextAccessor _workContextAccessor;
+        private readonly IHttpContextAccessor _httpContextAccessor;
+
+        public BaseAuthHandler(IWorkContextAccessor workContextAccessor, IHttpContextAccessor httpContextAccessor)
+        {
+            _workContextAccessor = workContextAccessor;
+            _httpContextAccessor = httpContextAccessor;
+        }
+
+        protected void AddCurrentUser(HttpRequestMessage request)
```

Comment:
Maybe make this class abstract, add `AddPlaformAuth` abstract method that should be implemented in the descendants, override `SendAsync` here with these lines
```
            AddCurrentUser(request);
            AddUserIp(request);

            AddPlaformAuth(request);
```
That there will be no code duplication and the addition of current user could not be forgotten in the descendants